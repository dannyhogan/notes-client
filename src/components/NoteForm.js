import React, { Component } from 'react';
import PropTypes from 'prop-types';

class NoteForm extends Component {

  static propTypes = {
    handleChange: PropTypes.func.isRequired,
    handleSubmit: PropTypes.func.isRequired,
    title: PropTypes.string.isRequired,
    note: PropTypes.string.isRequired
  }

  render() {
    return (
      <form onSubmit={this.props.handleSubmit}>
        <input type="text" value={this.props.title} name="titleInput" onChange={this.props.handleChange}></input>
        <input type="text" value={this.props.note} name="noteInput" onChange={this.props.handleChange}></input>
        <button type="submit">Submit</button>
      </form>
    );
  }
}

export default NoteForm;
