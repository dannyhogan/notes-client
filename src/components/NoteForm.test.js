import React from 'react';
import { shallow } from 'enzyme';
import NoteForm from './NoteForm';

describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(<NoteForm />);
    expect(wrapper).toMatchSnapshot();
  });
});
