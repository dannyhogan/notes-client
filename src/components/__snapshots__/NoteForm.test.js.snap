// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<form>
  <input
    name="titleInput"
    type="text"
  />
  <input
    name="noteInput"
    type="text"
  />
  <input
    name="authorInput"
    type="text"
  />
</form>
`;
